lib/internal_metric/global.rb

```diff
@@ -61,20 +64,34 @@ def collect
         { revision: @@version, version: Discourse::VERSION::STRING } => 1
       }
 
+      redis_primary_running = {}
+      redis_replica_running = {}
+
       redis_config = GlobalSetting.redis_config
-      redis_master_running = test_redis(:master, host: redis_config[:host], port: redis_config[:port], password: redis_config[:password], ssl: redis_config[:ssl])
-      redis_slave_running = 0
+      redis_primary_running[{ type: 'main' }] = test_redis(:master, host: redis_config[:host], port: redis_config[:port], password: redis_config[:password], ssl: redis_config[:ssl])
```

Comment:
It's a bit unusual to use a hash as a key in another hash, especially since just a string would do, but I guess it's for self-documentation? (and for potential expansion in the future?)

Reply:
Using hashes as keys is part of the prometheus-exporter gem API. In this case, it applies a "type=main" label to this value.